Tests/MapboxMapsTests/MapView/Integration Tests/OfflineManagerIntegrationTests.swift

```diff
@@ -0,0 +1,163 @@
+
+        /// Perform the download
+        let download = TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                                              loadOptions: tileRegionLoadOptions!) { _ in }
+        completion: { result in
+            switch result {
+            case .success(_):
+                XCTFail("Result reached success block, therefore download was not canceled")
+            case .failure(let error):
+                if error.localizedDescription == "Load was canceled" {
+                    downloadWasCancelled.fulfill()
+                } else {
+                    XCTFail("Download was not canceled")
+                }
+            }
+        }
+
+        download.cancel()
+
+        let expectations = [downloadWasCancelled]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testOfflineRegionCanBeDeleted() {
+
+        /// Expectations to be fulfilled
+        let downloadWasDeleted = XCTestExpectation(description: "Downloaded offline tiles were deleted")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                               loadOptions: tileRegionLoadOptions!) { _ in } completion: { _ in }
+
+        /// Waiting for the load tile to complete
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
+            TileStore.getInstance().removeTileRegion(forId: self.tileRegionId)
+
+            TileStore.getInstance().allTileRegions(completion: { result in
+                switch result {
+                case .success(let tileRegions):
+                    if tileRegions.count == 0 {
+                        downloadWasDeleted.fulfill()
+                    }
+                case .failure(let error):
+                    XCTFail("Error getting tile regions with error: \(error)")
+                }
+            })
+        }
+
+        let expectations = [downloadWasDeleted]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testMapCanBeLoadedWithoutNetworkConnectivity() {
+        /// Expectations to be fulfilled
+        let mapDidLoad = XCTestExpectation(description: "Map was loaded")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                               loadOptions: tileRegionLoadOptions!) { _ in } completion: { _ in }
+
+        NetworkConnectivity.getInstance().setMapboxStackConnectedForConnected(false)
+
+        self.mapView!.on(.resourceRequest) { event in
+            let eventElements = event.data as! [String: Any]
+
+            for element in eventElements {
+                if element.key == "data-source" && element.value as! String == "database" {
+                    mapDidLoad.fulfill()
+                }
+            }
+        }
```

Comment:
I wonder how we should handle this. As it is this will trigger the expectation as soon as 1 data-source / database event is emitted, even if other events come from the network.

Perhaps we need to listen to both `resourceRequest` and `mapLoaded`; in `resourceRequest` we could count the number of different data-sources, and in `mapLoaded` we check them. We'd expect `database` to be > 0 and `network` == 0.

Reply:
I've been thinking about this, and I feel like we need to think about what is the simplest test case. Here what we want to check is that if/when the map is going to load, where is it loading from. If there is a database as a data source, that tell's that it is not using a network to load which is what we are looking for. I don't think we need to check `mapLoaded` as well